### DISCUSSION TITLE
vercel dev requires a deployment

### DISCUSSION THREAD

**Discussion by giuseppeg** [vercel/vercel#4534] [Help] [answered]
A couple of months ago I could run `now dev` without any prior deployment to Now, yesterday I tried `vc dev` and it asked me to deploy the project else it would exit. Am I missing something or is this the new default?

**Comment by ctrlplusb** [comment]
As far as I can tell they want you to have a configured "project" - so that the settings, such as the dev command can be fetched and executed. Would be cool if a project could be initialised / configured in some other manner that didn't require an initial deploy.

**Comment by styfle** [comment]
> Would be cool if a project could be initialised / configured in some other manner that didn't require an initial deploy

Definitely! We are aware of this issue and are working on a way to link a project without deploying.

**Comment by styfle** [comment, marked as answer]
The canary channel has been updated with a new command, `vercel link`.

Additionally, `vercel dev` will ask to link to a project if it is not already.

Try it out now with `npm i -g vercel@latest`, thanks!

Update: There's also a discussion about running `vc dev` without project linking in #4925.

**Reply by giuseppeg** [reply]
@styfle I guess we were asking if there is a way to run `vercel dev` locally without linking or creating a deployment/project.

**Reply by styfle** [reply]
`vercel link` doesn't create a deployment so that part is solved.

`vercel dev` needs to link to a project in order to retrieve the dev command.

In the future, we might change this to allow running `vercel dev` without project linking.

**Comment by jadbox** [comment]
How do we run Vercel dev without any project to link to?
